source/device/tensorrt/trt_executor.cc

```diff
@@ -34,30 +34,90 @@ EXPORT_FINISH
 
 #include <NvInferRuntime.h>
 
+#include <limits>
+#include <fstream>
+
 
 static Logger gLogger(nvinfer1::ILogger::Severity::kERROR);
 
 
+
+void SaveEngine(const std::string& fileName, nvinfer1::ICudaEngine* mEngine)
+{
+    if (fileName == "")
+    {
+        std::cout << ("empty engine file name, skip save\n");
+        return;
+    }
+    if (mEngine != nullptr)
+    {
+        std::cout << ("save engine to %s...\n", fileName);
+        nvinfer1::IHostMemory* data = mEngine->serialize();
+        std::ofstream file;
+        file.open(fileName, std::ios::binary | std::ios::out);
+        if (!file.is_open())
+        {
+            std::cout << ("read create engine file %s failed\n", fileName);
+            return;
+        }
+        file.write((const char*)data->data(), data->size());
+        file.close();
+        data->destroy();
+    }
+    else
+    {
+        std::cout << ("engine is empty, save engine failed\n");
+    }
+}
+
+bool DeserializeEngine(const std::string& engineFile, nvinfer1::ICudaEngine** mEngine)
+{
+    std::ifstream in(engineFile.c_str(), std::ifstream::binary);
+    if (in.is_open())
+    {
+        std::cout << "deserialize engine from " << engineFile << std::endl;
+        auto const start_pos = in.tellg();
+        in.ignore(std::numeric_limits<std::streamsize>::max());
+        size_t bufCount = in.gcount();
+        in.seekg(start_pos);
+        std::unique_ptr<char[]> engineBuf(new char[bufCount]);
+        in.read(engineBuf.get(), bufCount);
```

Comment:
```suggestion
        std::vector<char> engineBuf;
        engineBuf.reserve(bufCount);
        engineBuf.insert(engineBuf.begin(), std::istreambuf_iterator<char>(in), std::istreambuf_iterator<char>());
```

no need to use smart point